crates/biome_js_analyze/src/lint/nursery/no_useless_length_check.rs

```diff
@@ -0,0 +1,366 @@
+    let Some(parent) = node.syntax().parent() else {
+        return false;
+    };
+    parent
+        .ancestors()
+        .find_map(|ancestor| {
+            if let Some(logical_exp) = JsLogicalExpression::cast(ancestor.clone()) {
+                return logical_exp
+                    .operator_token()
+                    .ok()
+                    .map(|token| token.kind() == operator)
+                    .or(Some(false));
+            }
+            (!JsParenthesizedExpression::can_cast(ancestor.kind())).then_some(false)
+        })
+        .unwrap_or(false)
+}
+
+/// Extract the expressions that perform length comparisons corresponding to the errors you want to check.
+fn get_comparing_length_exp(
+    binary_exp: &JsBinaryExpression,
+    expect_error: &ErrorType,
+) -> Option<AnyJsExpression> {
+    let left = binary_exp.left().ok()?;
+    let operator = binary_exp.operator().ok()?;
+    let right = binary_exp.right().ok()?;
+
+    // Check only when the number appears on the right side according to the original rules.
+    // We assume that you have already complied with useExplicitLengthCheck
+    let compare_exp = left;
+    let value_exp = right;
+
+    let AnyJsExpression::JsStaticMemberExpression(member_exp) = compare_exp else {
+        return None;
+    };
+    let target = member_exp.object().ok()?;
+    let member = member_exp.member().ok()?;
+    if member.text() != "length" || member_exp.is_optional_chain() {
+        return None;
+    }
+    let AnyJsExpression::AnyJsLiteralExpression(AnyJsLiteralExpression::JsNumberLiteralExpression(
+        literal,
+    )) = value_exp
+    else {
+        return None;
+    };
+    let number = literal.as_number()?.round() as i64;
+    // .length === 0
+    if matches!(expect_error, ErrorType::UselessLengthCheckWithEvery)
+        && literal.syntax().text_trimmed() == "0"
+        && (operator == JsBinaryOperator::StrictEquality || operator == JsBinaryOperator::LessThan)
+    {
+        return Some(target);
+    }
+    // .length !== 0
+    if matches!(expect_error, ErrorType::UselessLengthCheckWithSome)
+        && (literal.syntax().text_trimmed() == "0"
+            && (operator == JsBinaryOperator::StrictInequality
+                || operator == JsBinaryOperator::GreaterThan)
+            || number > 0 && operator == JsBinaryOperator::StrictEquality
+            || literal.syntax().text_trimmed() == "1"
+                && operator == JsBinaryOperator::LessThanOrEqual)
+    {
+        return Some(target);
+    }
+    None
+}
+
+pub type Replacer = (JsLogicalExpression, AnyJsExpression, TextRange);
+
+/// Search for logical expressions and list expressions that compare to 0 and Array APIs (`.some()`, `.every()`).
+fn search_logical_exp(
+    any_exp: &AnyJsExpression,
+    replacer: Option<Replacer>,
+    expect_error: &ErrorType,
+    comparing_zeros: &mut HashMap<String, Vec<Replacer>>,
+    array_tokens_used_api: &mut HashSet<String>,
```

Comment:
There are too many arguments here, I think. But it's fine. However, you should document them and explain what they are. What's `any_exp`, for example? It's difficult to understand what it is, especially in relation to the name to `search_local_exp`. Maybe you want to give a better name, like `expression_to_search`.

Also, `replacer` is a bit cryptic. Usually, you should try to give your function a name based on its business logic, not on its type.